internal/pkg/deploy/cloudformation/stack/name.go

```diff
@@ -50,3 +53,11 @@ func NameForAppStack(app string) string {
 func NameForAppStackSet(app string) string {
 	return fmt.Sprintf("%s-infrastructure", app)
 }
+
+// NameForPipeline returns the stack name for a pipeline, depending on whether it has been deployed using the legacy scheme.
+func NameForPipeline(app string, pipeline string, isLegacy bool) string {
+	if isLegacy {
+		return pipeline
+	}
+	return fmt.Sprintf(fmtPipelineNamespaced, app, pipeline)
```

Comment:
do we validate anywhere that the length of this combo is not longer than 128 chars? or even better can we automatically balance things out automatically:
```go
// input: pipeline-{appName that's 100 chars}-{pipelineName that's 100 chars}

func shorten(s string, sampleSize int) string {
   if len(s) < sampleSize {
       return s
   }
   return s[:sampleSize]
}

return fmt.Sprintf(fmtPipelineNamespaced, shorten(app, 59 /* (128 - 10)/2 */), shorten(pipeline, 59))
```